pontoon/pretranslation/tasks.py

```diff
@@ -2,18 +2,22 @@
 
 from django.db.models import Q, CharField, Value as V
 from django.db.models.functions import Concat
+from django.conf import settings
 
 from pontoon.base.models import Entity, TranslatedResource, Translation
 from pontoon.pretranslation.pretranslate import (
     get_translations,
     update_changed_instances,
 )
+from pontoon.base.tasks import PontoonTask
+from pontoon.sync.core import serial_task
 
 
 log = logging.getLogger(__name__)
 
 
-def pretranslate(project, locales=None, entities=None):
+@serial_task(settings.SYNC_TASK_TIMEOUT, base=PontoonTask, lock_key="project={0}")
+def pretranslate(self, project, locales=None, entities=None):
```

Comment:
I haven't tested this yet, but I wonder if this works as expected when called from `pontoon/sync/tasks.py`.

Reply:
I don't remember exactly, but I might have tested both cases before opening the PR during All hands. I will do it once again.

Reply:
@mathjazz : It is working as expected for me in both the cases.